### PR TITLE
libc/stdio: Fix bug when CONFIG_LIBC_NUMBERED_ARGS is enabled

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- libc/stdio: Add _s suffix for struct arg in lib_libvsprintf.c 
- libc/stdio: Skip fetch double argument if !CONFIG_LIBC_FLOATINGPOINT && CONFIG_LIBC_NUMBERED_ARG
- libc/stdio: Don't fetch width/prec modifier in parser phase
- libc/stdio: Make %p[V|S|s] work with CONFIG_LIBC_NUMBERED_ARGS
- boards/sim/nsh: Enable ONFIG_LIBC_NUMBERED_ARGS and ONFIG_SCHED_BACKTRACE

## Impact
printf series function with CONFIG_LIBC_NUMBERED_ARGS == y

## Testing
sim:nsh

### CODE REVIEW

File: libs/libc/stdio/lib_libvsprintf.c [hunk 1]
```diff
@@ -1176,6 +1123,60 @@ static int vsprintf_internal(FAR struct lib_outstream_s *stream,
...
+#ifdef CONFIG_ALLSYMS
+                  case 'S':
+                  case 's':
+                    {
+                      FAR const struct symtab_s *symbol;
+                      FAR void *addr = (FAR void *)x;
+                      size_t symbolsize;
+
+                      symbol = allsyms_findbyvalue(addr, &symbolsize);
+                      if (symbol)
```
Comment (pkarashchenko) [inline libs/libc/stdio/lib_libvsprintf.c:1153]:
```suggestion
                      if (symbol != NULL)
```

Comment (xiaoxiang781216) [reply libs/libc/stdio/lib_libvsprintf.c]: Done.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

File: libs/libc/stdio/lib_libvsprintf.c [hunk 2]
```diff
@@ -1176,6 +1123,60 @@ static int vsprintf_internal(FAR struct lib_outstream_s *stream,
               break;
 
             case 'p':
+              c = fmt_char(fmt);
+              switch (c)
+                {
+                  case 'V':
+                    {
+                      FAR struct va_format *vaf = (FAR void *)x;
```
Comment (pkarashchenko) [inline libs/libc/stdio/lib_libvsprintf.c:1131]: I see
```
#ifndef CONFIG_LIBC_LONG_LONG
          unsigned long x;
#else
          unsigned long long x;
```
So the question is: Is `unsigned long` is capable to carry full pointer value?

Comment (xiaoxiang781216) [reply libs/libc/stdio/lib_libvsprintf.c]: Why not? x will be unsigned long long in most arch, it's enough to contain all possible pointer value.

**Pull request merged**